columnflow/production/cms/pdf.py

```diff
@@ -30,24 +30,49 @@
     # only run on mc
     mc_only=True,
 )
-def pdf_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
+def pdf_weights(
+    self: Producer,
+    events: ak.Array,
+    outlier_threshold: float = 0.5,
+    outlier_mode: str = "keep",
+    outlier_message_type: str = "warning",
+    **kwargs,
+) -> ak.Array:
     """
     Producer that determines the pdf up and down variations on an event-by-event basis.
     This producer assumes that the nominal entry is always the first LHEPdfWeight value and
     that the nominal weight is already included in the LHEWeight.
     Can only be called with MC datasets.
 
+    The *outlier_mode* defines the prodcedure of how to handle events with a pdf uncertainty above
+    the *outlier_threshold*. Supported modes are:
+
+        - ``"keep"``: events are kept unmodified
+        - ``"remove"``: pdf weight nominal/up/down are all set to 0, essentially removing the event
+
+    The *outlier_message_type* allows to define the type of message that is given for datasets with outliers:
+
+        - ``"ignore"``: no message is given
+        - ``"info"``: a `logger.info` message is given
+        - ``"debug"``: a `logger.debug` message is given
+        - ``"warning"``: a `logger.warning` message is given
+        - ``"raise"``: an Exception is raised
+
```

Comment:
I think outlier mode and message type have some overlap in their meaning, e.g. `raise` should be a mode.

How about

```
The *outlier_mode* defines the procedure of how to handle events with a pdf uncertainty above the *outlier_threshold*. Supported modes are:

    - ``"ignore"``: events are kept unmodified, no message is printed
    - ``"warning"``: events are kept unmodified, a warning is printed
    - ``"remove"``: pdf weight nominal/up/down are all set to 0, a warning is printed  
    - ``"raise"``: an exception is raised
```

This seems a bit more streamlined to me.

Reply:
In my opinion, these are two separate things we might want to customize, one parameter for verbosity and one for functionality (how to handle the outlier events). With your suggestion, there would be no option to "remove" the outlier events but also print no message.

Reply:
Just to restart the conversation in this thread: I agree with @mafrahm, the handling of the events and the logging/information for the users can (and maybe should) be to separate things. Maybe just use one `outlier_log_mode` and an `outlier_action` as input parameters? So something like this:

```
The *outlier_action* defines the procedure of how to handle events with a pdf uncertainty above the *outlier_threshold*. Supported modes are:

    - ``"ignore"``: events are kept unmodified
    - ``"remove"``: pdf weight nominal/up/down are all set to 0, a warning is printed  
    - ``"raise"``: an exception is raised
    
 Additionally, the verbosity of the procedure can be set with *outlier_log_mode*, which offers the following options:

    - ``"info"``: a `logger.info` message is given
    - ``"debug"``: a `logger.debug` message is given
    - ``"warning"``: a `logger.warning` message is given

```

This way we could have silent handling of outlier weights, or a more verbose documentation of such incidents in different logger streams. What do you think @mafrahm @riga ?

Reply:
Hi @pkausw, I agree with your proposal, with two exceptions:
- `outlier_action = remove` should not print a warning per default but based on the *outlier_log_mode*
- the `outlier_log_mode` should offer a mode to completely ignore the message (e.g. None or False)

Reply:
sounds good to me :+1:

Reply:
To me as well :+1: